src/sentry/static/sentry/app/views/organizationEventsV2/transactionView/dividerHandlerManager.tsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+
+import {
+  rectOfContent,
+  clamp,
+  toPercent,
+  UserSelectValues,
+  setBodyUserSelect,
+} from './utils';
+
+// divider handle is positioned at 50% width from the left-hand side
+const DEFAULT_DIVIDER_POSITION = 0.5;
+
+const selectDividerLines = (transform: (dividerDOM: HTMLDivElement) => void): void => {
+  document
+    .querySelectorAll<HTMLDivElement>("[data-test-id='divider-line']")
+    .forEach(transform);
+};
+
+const selectGhostDividerLines = (
+  transform: (dividerDOM: HTMLDivElement) => void
+): void => {
+  document
+    .querySelectorAll<HTMLDivElement>("[data-test-id='divider-line-ghost']")
+    .forEach(transform);
+};
+
+export type DividerHandlerManagerChildrenProps = {
+  dividerPosition: number;
+  setHover: (nextHover: boolean) => void;
+  onDragStart: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+};
+
+type StateType = {
+  dividerPosition: number; // between 0 and 1
+};
+
+const DividerManagerContext = React.createContext<DividerHandlerManagerChildrenProps>({
+  dividerPosition: DEFAULT_DIVIDER_POSITION,
+  onDragStart: () => {},
+  setHover: () => {},
+});
+
+type PropType = {
+  children: React.ReactNode;
+  interactiveLayerRef: React.RefObject<HTMLDivElement>;
+};
+
+export class Provider extends React.Component<PropType, StateType> {
+  state: StateType = {
+    dividerPosition: DEFAULT_DIVIDER_POSITION,
+  };
+
+  previousUserSelect: UserSelectValues | null = null;
+  dividerHandlePosition: number = DEFAULT_DIVIDER_POSITION;
+  isDragging: boolean = false;
+
+  hasInteractiveLayer = (): boolean => {
+    return !!this.props.interactiveLayerRef.current;
+  };
+
+  setHover = (nextHover: boolean) => {
+    if (this.isDragging) {
+      return;
+    }
+
+    selectDividerLines(dividerDOM => {
+      if (nextHover) {
+        dividerDOM.classList.add('hovering');
+        return;
+      }
+
+      dividerDOM.classList.remove('hovering');
+    });
+  };
+
+  onDragStart = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    if (this.isDragging || event.type !== 'mousedown' || !this.hasInteractiveLayer()) {
+      return;
+    }
+
+    event.stopPropagation();
+
+    // prevent the user from selecting things outside the minimap when dragging
+    // the mouse cursor outside the minimap
+
+    this.previousUserSelect = setBodyUserSelect({
+      userSelect: 'none',
+      MozUserSelect: 'none',
+      msUserSelect: 'none',
+    });
+
+    // attach event listeners so that the mouse cursor does not select text during a drag
+    window.addEventListener('mousemove', this.onDragMove);
```

Comment:
It might be worth adding a (e.g. invisible) element or something to add the event listener to so React will properly cleanup on unmount. There might be a chance of them not being removed properly.

Reply:
I've added the missing `componentWillUnmount()`.